### PR TITLE
Refactor C++ code and prepare python integration

### PR DESCRIPTION
**Pull request by dendisuhubdy**
- [x] adding include folder structure
- [x] adjusting graphs functions in different classes
- [x] adding RL base class

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -1,224 +1,7 @@
-#include <iostream>
-#include <queue>
-#include <algorithm>
-#include <string.h>
-#include <set>
-#include <climits>
-#include <fstream>
+#include "stdafx.h"
```
Comment (dmitriy-serdyuk) [inline src/main.cpp]: do you need to include `graphs.h`?

Comment (dendisuhubdy) [reply src/main.cpp]: Exactly sorry I forgot @dmitriy-serdyuk

Comment (dendisuhubdy) [reply src/main.cpp]: Adjusted in the following commit

**Comment by dmitriy-serdyuk** [comment]
Dendi, I propose to split this PR into three:
  - Refactor C++ code and prepare to python integration
  - Create C++ - python interface
  - Add RL stuff

**Comment by dendisuhubdy** [comment]
@dmitriy-serdyuk ok then, if you can review this. We can probably merge it then submit the C++ python interface up.

File: src/main.cpp [hunk 2]
```diff
@@ -1,224 +1,10 @@
...
-				dist[i][j] = bfs(i,j,graph);
-			} else {
-				dist[i][i] = 0;
-			}
-		}
-	}
-}
 
 //A very simplified QASM parser
 void read_qasm(std::ifstream& infile) {
```
Comment (dmitriy-serdyuk) [inline src/main.cpp]: You are using namespace std already, do you need to reference it? I am not sure what C++ style guides say about this.

**Review by dmitriy-serdyuk — CHANGES_REQUESTED** [review]
We need to split mapping and optimization. As far as I understand, mapping starts at https://github.com/dendisuhubdy/quantum_compiler_optim/pull/1/files#diff-7ec3c68a81efff79b6ca22ac1f1eabbaR373

File: src/main.cpp (hunk 2 above)
Comment (dendisuhubdy) [reply src/main.cpp]: I don't you're correct, in fact the good practice is not to use `using namespace std`, because it can cause conflicts.

Comment (dmitriy-serdyuk) [reply src/main.cpp]: 👍

**Comment by dendisuhubdy** [comment]
@dmitriy-serdyuk adjusted the source code to split between mapping and optimization.

**Pull request closed without merging**